src/PrintCore/PrintCore.cs

```diff
@@ -0,0 +1,775 @@
+		internal IntPtr handle;
+
+		internal PMPrintCoreBase ()
+		{
+			// For delayed initialization cases.
+		}
+		
+		internal PMPrintCoreBase (IntPtr handle) : this (handle, false) {}
+		internal PMPrintCoreBase (IntPtr handle, bool owns)
+		{
+			if (!owns)
+				NativeInvoke.PMRetain (handle);
+			this.handle = handle;
+		}
+
+		~PMPrintCoreBase ()
+		{
+			Dispose (false);
+		}
+
+		public void Dispose ()
+		{
+			Dispose (true);
+			GC.SuppressFinalize (this);
+		}
+
+		public IntPtr Handle => handle;
+
+		protected virtual void Dispose (bool disposing)
+		{
+			if (handle != IntPtr.Zero){
+				NativeInvoke.PMRelease (handle);
+				handle = IntPtr.Zero;
+			}
+		}
+	}
+
+	public class PMPrintException : Exception {
+		public PMPrintException (PMStatusCode code) : base (code.ToString ()){}
+	}
+	
+	public class PMPrintSession : PMPrintCoreBase {
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMCreateSession (out IntPtr session);
+
+		internal PMPrintSession (IntPtr handle, bool owns) : base (handle, owns) {}
+		
+		public PMPrintSession ()
+		{
+			IntPtr value;
+			var code = PMCreateSession (out value);
+			if (code == PMStatusCode.Ok)
+				handle = value;
+			else
+				throw new PMPrintException (code);
+		}
+
+		public static PMStatusCode TryCreate (out PMPrintSession session)
+		{
+			IntPtr value;
+			var code = PMCreateSession (out value);
+			if (code == PMStatusCode.Ok){
+				session = new PMPrintSession (value, true);
+				return PMStatusCode.Ok;
+			}
+			session = null;
+			return code;
+		}
+
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionError (IntPtr handle);
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionSetError (IntPtr handle, PMStatusCode code);
+		
+		public PMStatusCode SessionError {
+			get {
+				return PMSessionError (handle);
+			}
+			set {
+				PMSessionSetError (handle, value);
+			}
+		}
+		
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionDefaultPrintSettings (IntPtr session, IntPtr settings);
+
+		public void AssignDefaultSettings (PMPrintSettings settings)
+		{
+			if (settings == null)
+				throw new ArgumentNullException (nameof (settings));
+			PMSessionDefaultPrintSettings (handle, settings.handle);
+		}
+
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionDefaultPageFormat (IntPtr session, IntPtr pageFormat);
+		
+		public void AssignDefaultPageFormat (PMPageFormat pageFormat)
+		{
+			if (pageFormat == null)
+				throw new ArgumentNullException (nameof (pageFormat));
+			PMSessionDefaultPageFormat (handle, pageFormat.Handle);
+		}
+		
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionCreatePrinterList (IntPtr printSession, out IntPtr printerListArray, out int index, out IntPtr printer);
+
+		internal static string [] FetchArray (IntPtr cfArrayString, bool owns)
```

Comment:
Does not seem generic enough, it fetches an array of strings.   

I will replace with the NSArray tool that does the same.